src/main/scala/org/scanet/optimizers/SGD.scala

```diff
@@ -0,0 +1,20 @@
+package org.scanet.optimizers
+
+import org.scanet.core.{Output, Tensor, TensorType}
+import org.scanet.math.Numeric
+import org.scanet.math.syntax._
+
+case class SGD(rate: Double = 0.01, momentum: Double = 0.0, nesterov: Boolean = false) extends Algorithm[Tensor[Float]] {
+
+  override def delta[A: Numeric : TensorType, B: Numeric : TensorType](f: TensorFunction[A, B], initArg: Output[A], prevState: Tensor[Float]): (Output[Float], Tensor[Float]) = {
+    val g = f.grad(initArg)
+    val m = momentum.toFloat.const
+    val r = rate.toFloat.const
+    val v = r * g - m * prevState.const
+    val velocity = if (nesterov) r * g - m * v else v
+    val res = velocity.eval
+    (res.const, res)
```

Comment:
I guess the state is a `v` and not `velocity`

Reply:
thanks! fixed